File: python/1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree.py
#Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:
        def helper(orig, clone, target):
            if orig == None or clone == None:
                return None
            elif orig == target:
                return clone
            leftResult = helper(orig.left,clone.left,target)
            if leftResult == None:
                return helper(orig.right,clone.right,target)
            return leftResult
        return helper(original, cloned, target)

File: java/1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {

    public class TreeNode {
      int val;
      TreeNode left;
      TreeNode right;
      TreeNode(int x) { val = x; }
    }

    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        return helper(original, cloned, target);
        
    }
    public TreeNode helper(TreeNode orig, TreeNode clone, TreeNode target){
        if(orig == null || clone == null){
            return null;
        }
        if(orig == target){
            return clone;
        }
        TreeNode leftResult = helper(orig.left, clone.left, target);
        return (leftResult == null) ? helper(orig.right,clone.right, target) : leftResult;
    }
    
}